pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -107,30 +107,57 @@ func (l *Lifecycle) driverExists(obj *v3.KontainerDriver) bool {
 
 func (l *Lifecycle) download(obj *v3.KontainerDriver) (*v3.KontainerDriver, error) {
 	driver := drivers.NewKontainerDriver(obj.Spec.BuiltIn, obj.Name, obj.Spec.URL, obj.Spec.Checksum)
-	err := driver.Stage()
+
+	if !driver.Exists() {
+		v3.KontainerDriverConditionDownloaded.Unknown(obj)
+		v3.KontainerDriverConditionInstalled.Unknown(obj)
+	}
+	// update status
+	obj, err := l.kontainerDrivers.Update(obj)
```

Comment:
This should be in `if !driver.Exists(){` block, you don't have to update anytime.

Reply:
I move the operation of update here, so that there aren't so many update operation in this func.